best-cigars-guide/scripts/delayed.js

```diff
@@ -5,3 +5,43 @@ import { sampleRUM } from './aem.js';
 sampleRUM('cwv');
 
 // add more delayed functionality here
+
+function onError(tagName) {
+  /* eslint no-console: ["error", { allow: ["error"] }] */
+  console.error(`The ${tagName} script failed to load.`);
+}
+
+function onGALoad() {
+  /* global dataLayer */
+  window.dataLayer = window.dataLayer || [];
+  function gtag() {
+    // eslint-disable-next-line prefer-rest-params
+    dataLayer.push(arguments);
+  }
+  gtag('js', new Date());
+  gtag('config', 'GT-M6QM788');
+  gtag('set', 'linker', { domains: ['www.famous-smoke.com'] });
+}
+
+function loadGoogleAnalytics() {
+  // Load the Google Analytics library
+  const tag = document.createElement('script');
+  tag.async = true;
+  tag.src = 'https://www.googletagmanager.com/gtag/js?id=GT-M6QM788';
+  document.head.appendChild(tag);
+  // Configuration script
+  tag.onload = onGALoad;
+  tag.onerror = onError('Google Analytics');
```

Comment:
onError called like this will always be triggered since a param is being passed in, even if the script loads successfully. See the JS error occurring in the live version of this even with GTM successfully loading:

<img width="993" alt="image" src="https://github.com/famous-smoke/best-cigars-guide/assets/11136895/5d5dad49-d273-497d-b27b-5d8fb1feef3b">


This can be wrapped in a function to produce the desired behavior:

```javascript
  tag.onerror = () => {
    onError('Google Analytics');
  };
```